File: src/main/java/at/xirado/bean/command/slashcommands/TestCommand.java
package at.xirado.bean.command.slashcommands;

import at.xirado.bean.command.SlashCommand;
import at.xirado.bean.command.SlashCommandContext;
import net.dv8tion.jda.api.entities.Member;
import net.dv8tion.jda.api.events.interaction.command.SlashCommandInteractionEvent;
import net.dv8tion.jda.api.interactions.commands.OptionType;
import net.dv8tion.jda.api.interactions.commands.build.Commands;
import org.jetbrains.annotations.NotNull;
import org.jetbrains.annotations.Nullable;

public class TestCommand extends SlashCommand
{
    public TestCommand()
    {
        setCommandData(Commands.slash("test", "this command is only for test purposes")
                .addOption(OptionType.BOOLEAN, "ephemeral", "if this message is ephemeral", true)
        );
        setGlobal(false);
        setEnabledGuilds(815597207617142814L);
    }

    @Override
    public void executeCommand(@NotNull SlashCommandInteractionEvent event, @NotNull SlashCommandContext ctx)
    {
        Member member = event.getMember();
        long balance = ctx.getGuildData().getBalance(member.getIdLong());
        event.reply("Your balance is: $" + balance).queue();
    }
}

File: src/main/kotlin/at/xirado/bean/interaction/command/slash/developer/TestCommand.kt
package at.xirado.bean.interaction.command.slash.developer

import at.xirado.bean.Application
import at.xirado.bean.i18n.LocalizationManager
import at.xirado.bean.interaction.CommandFlag
import at.xirado.bean.interaction.SlashCommand
import at.xirado.bean.util.replyError
import dev.minn.jda.ktx.await
import dev.minn.jda.ktx.interactions.getOption
import net.dv8tion.jda.api.events.interaction.command.CommandAutoCompleteInteractionEvent
import net.dv8tion.jda.api.events.interaction.command.SlashCommandInteractionEvent
import net.dv8tion.jda.api.interactions.commands.OptionType

class TestCommand(private val application: Application): SlashCommand("test", "this is a test") {

    init {
        setEnabledGuilds(*application.config.devGuilds.toLongArray())
        addCommandFlags(CommandFlag.DEVELOPER_ONLY)
        option(type = OptionType.STRING, "locale", "the locale to test against", required = true, autoComplete = true)
    }

    override suspend fun execute(event: SlashCommandInteractionEvent) {
        val option = event.getOption<String>("locale")!!
        val lang = LocalizationManager.getForLanguageTag(option)?: return kotlin.run {
            event.replyError("Locale $option was not found!").await()
        }

        val user = event.user.asTag
        val reason = "he codes in Kotlin"

        val template = lang.get("general.info")!!
        val filled = lang.get("general.info", "user" to user, "reason" to reason)!!
        event.reply("Locale: $option\nTemplate: $template\nFilled out: $filled").await()
    }

    override suspend fun onAutoComplete(event: CommandAutoCompleteInteractionEvent) {
        event.replyChoiceStrings(LocalizationManager.LANGUAGES.filter { it.startsWith(event.focusedOption.value, true) } .map { it.replace(".yml", "") }.toList()).await()
    }
}